### ISSUE TITLE
"Check Geometric" core plugin not work for finding gaps

### ISSUE THREAD

**Issue by onsummer** [qgis/QGIS#41380]
I create a memory multipolygon layer, and create a polygon with 2 parts -- they have a near 0.3 meters gaps.

## **How to Reproduce**

1. create a memory multipolygon layer and create a multipolygon with 2 parts (spatial reference: wkid 3857), with a near 0.3 meters gaps
2. run the geometry check plugin, select the layer, set Topology checks - Check for gaps smaller than - 0.5 (map unit: meter)
3. then no result.

### data:

![image](https://user-images.githubusercontent.com/26168955/107072829-8042c100-6821-11eb-8e85-6dcaeee3bc37.png)

### set options:

![Snipaste_2021-02-06_02-13-43](https://user-images.githubusercontent.com/26168955/107072393-03afe280-6821-11eb-8ceb-10c94bca4975.jpg)

### result (after I press the Run button): 

![image](https://user-images.githubusercontent.com/26168955/107072473-2215de00-6821-11eb-9cde-e32d660c568d.png)

## **QGIS and OS versions**
QGIS versions: 3.16.3-Hannover
QGIS code version: 94ac9f21b8
GDAL/OGR version: 3.1.4
OS version: windows 10

**Comment by onsummer** [comment]
there is the data in geojson format:

``` json
{
  "type": "FeatureCollection",
  "name": "111",
  "crs": {
    "type": "name",
    "properties": {
      "name": "urn:ogc:def:crs:EPSG::3857"
    }
  },
  "features": [
    {
      "type": "Feature",
      "properties": {},
      "geometry": {
        "type": "MultiPolygon",
        "coordinates": [
          [
            [
              [
                12573625.430904950946569,
                2502896.047371748369187
              ],
              [
                12603622.656450632959604,
                2625787.906865348573774
              ],
              [
                12705098.571768241003156,
                2586649.726611960213631
              ],
              [
                12666208.13187269680202,
                2469150.950757336337119
              ],
              [
                12573625.430904950946569,
                2502896.047371748369187
              ]
            ]
          ],
          [
            [
              [
                12603621.798861145973206,
                2625785.391819250769913
              ],
              [
                12573624.573315463960171,
                2502893.532325650565326
              ],
              [
                12492734.600205007940531,
                2532376.793552826158702
              ],
              [
                12524442.068797780200839,
                2656324.170779117848724
              ],
              [
                12603621.798861145973206,
                2625785.391819250769913
              ]
            ]
          ]
        ]
      }
    }
  ]
}
```

**Comment by gioman** [comment]
@onsummer I guess that being a multipart polygon (so just 1 feature) technically there is no space/gap.

**Comment by onsummer** [comment]
@gioman well... in some cases, a piece of data is produced by a team, and they perhaps forget the small gaps when fusing data v😂🤦‍♂️

**Comment by gioman** [comment]
>  well... in some cases, a piece of data is produced by a team, and they perhaps forget the small gaps when fusing data

@onsummer yes I understand, but still that is not a gap between two different features, which is what the geometry checker looks for (I guess).

**Comment by gioman** [comment]
You can close those gaps with Processing/GRASS v.clean tool. You will have to rebuild the multipart features after it.

**Comment by uclaros** [comment]
Actually check geometries does find gaps between parts. This "space between islands" is not considered a gap by check geometries. To qualify as a gap it must be definable by a polygon with finite area, so it must be enclosed by features, think of remaining holes after a dissolve. The hint here is that you ask for *gaps smaller than (map units sqr)*, while your not-found-gap has no finite area!

**Comment by gioman** [comment]
> This "space between islands" is not considered a gap by check geometries. To qualify as a gap it must be definable by a polygon with finite area, so it must be enclosed by features

@uclaros of course :)

**Comment by uclaros** [comment]
@gioman so I see no bug here. Could be turned to a feature request to detect "space between islands" as gaps like arcgis topology does though.

**Comment by gioman** [comment]
> @gioman so I see no bug here. Could be turned to a feature request to detect "space between islands" as gaps like arcgis topology does though.

@uclaros done
